core/lib/spree/core/search/base.rb

```diff
@@ -57,16 +57,18 @@ def add_eagerload_scopes(scope)
         end
 
         def add_search_scopes(base_scope)
-          if @properties[:search]
-            @properties[:search].each do |name, scope_attribute|
-              scope_name = name.to_sym
-              if base_scope.respond_to?(:search_scopes) && base_scope.search_scopes.include?(scope_name.to_sym)
-                base_scope = base_scope.send(scope_name, *scope_attribute)
-              else
-                base_scope = base_scope.merge(Spree::Product.ransack({ scope_name => scope_attribute }).result)
-              end
+          return base_scope unless @properties[:search].presence.respond_to?(:each_pair)
+
+          @properties[:search].each_pair do |name, scope_attribute|
```

Comment:
It's a bit hidden in the git diff here, but I changed `each` into `each_pair` to reflect the new `respond_to?` check we added to this method. Leaving `each` would work anyway, but it's a bit strange to check that the object responds to a method and then use another, even if they are aliased. I couldn't have used `respond_to?(each)` because even other kinds of enumerables (like arrays) would be valid at that point while the rest of the code expects a key/value enumerable.